### DISCUSSION TITLE
More tools for testers

### DISCUSSION THREAD

**Discussion by Leroy8888** [HelixNebulaStudio/RiseOfTheDead#39] [Ideas]
I've had this thought for a long time, as tester it's sometimes pretty difficult to replicate or investigate / view certain issues as the version between live branch and dev branch differ significantly. I've also had many occassions where I was unable to do so due to the different versions in the 2, despite the issue or error being present in live branch. The issues which were possible to replicate usually cost a lot of time and effort for no gain, compared to when one simply can type 2 commands in dev branch.

As a suggestion I would like to propose an option for testers to use dev branch commands in main branch, but under one circumstance. Testers would have to pick between 2 slots, their regular slot with all their data and without (all) command permissions and one that is the equivalent of a data reset with all commands allowed (test file*).

When using the test file* that allows commands and other properties like in dev branch, the tester would be teleported to a different world / place that specifically allows the uses of these commands. This world would have no players, but allows other testers to join if they were to select the test file*. It would also be the same version as live branch. Testers wouldn't be able to transfer items from this file to their regular file and anything saved in this file including stats would be seperate from their regular file.

This would function as an option for testers to quickly replicate issues and other concerns without too much fuss, while making dev branch act as a version with updated concepts ready to be tested by either regular members or testers.

**Comment by MXKhronos** [comment]
As much as I understand how this would be useful. There's a lot of backend issues with how data is separated which is why I don't really dare messing with that part of the system. From your profile data that saves unlocked collectibles to item unlockables then to your save data containing storages, mission data and other resettable data. Then there's faction data which is a shared data file that's used by all members of a faction to live data that stores world and server information etc..

It's not exactly safe to try to isolate the part of the data for testing purposes because there can be a lot of mixed up even if I were to attempt to implement a simple system, it could mess up other part of the data, they are pretty intertwined. 

I should mention however that the more time passes without an update, the more dev branch and live branch differ and if a bug exist in main, it may be fixed in dev. I think it's better to temporarily conclude the dev branch's result for any issues as it's the most up to date. Until the update is complete and pushed onto live-branch, bugs in live branch can be considered as irrelevant if it couldn't be reproduced in dev. As in now, where db differs greatly, live branch will just have to wait til the update is released before posting any live branch bug reports.

What I can help with are tools for dev branch if you wish to perform a task that you currently do not have the ability to.

**Comment by Leroy8888** [comment]
Alright, I already figured that tinkering with data on main would be a bit risky.

A command to change the game to a certain version would be nice. For example where it would create a new temporary server when doing so. If this is doable at least.

I did also send a long message with one partly in it to the HNS account a bit ago, which was mixed with a member rank request for the wiki. I'll just seperate the one sentence that was refering to the command;

''The question here would be whether it would be possible to add some sort of feature to dev branch that allows users to switch or update their weapons to certain models or include certain animations of past versions. It would also be great if those were included from legacy times, although most of the emphasis is on the current models and animations in main branch.''

**Comment by MXKhronos** [comment]
Oh, that also has its own issue relating to how old animations can't be played on new weapon rigs depending on how the tool is rigged up. Basically, can't properly play old animations without the old rig, (e.g. deagle remodel) and old model and new model cannot exist at the same time.

**Comment by Leroy8888** [comment]
I did actually have a request once more. 

Would it be possible to create a new world in dev branch with the old harbor map that could be accessed by using the /travel command?
For example with /travel TheHarborV1 or TheHarborOld.

Also as a lil addition just in case someone in main branch decides to do something funny while it is included, maybe it should have the same mechanic as BioX where it would kick and or ban anyone who tries to somehow enter it via main branch.
